server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Keywords.java

```diff
@@ -0,0 +1,167 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import javax.mail.Flags;
+
+import org.apache.james.mailbox.FlagsBuilder;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+public class Keywords {
+    public interface KeywordsValidator {
+        void validate(Set<Keyword> keywords);
+    }
+
+    private FlagsBuilder combiner(FlagsBuilder firstBuilder, FlagsBuilder secondBuilder) {
+        return firstBuilder.add(secondBuilder.build());
+    }
+
+    private FlagsBuilder accumulator(FlagsBuilder accumulator, Keyword keyword) {
+        return accumulator.add(keyword.asFlags());
+    }
+
+    public static class KeywordsFactory {
+        private KeywordsValidator validator;
+        private Predicate<Keyword> filter;
+
+        private KeywordsFactory() {
+            validator = keywords -> {};
+            filter = keyword -> true;
+        }
+
+        public KeywordsFactory throwOnAddDraft() {
```

Comment:
How does keyword knows if it is a add or a remove?

Akka your method name express the caller intent.

**throwOnDraft** is better.